handler/program.go

```diff
@@ -117,3 +117,85 @@ func CreateProgram(cc echo.Context) error {
 	return c.JSON(http.StatusCreated, &p)
 
 }
+
+func DeleteProgram(cc echo.Context) error {
+	c := cc.(*db.DBContext)
+	// acquire parameters via anonymous struct.
+	var req struct {
+		UID string `json:"uid"`
+		PID string `json:"pid"`
+	}
+	if err := httpext.RequestBodyTo(c.Request(), &req); err != nil {
+		return c.String(http.StatusInternalServerError, errors.Wrap(err, "failed to read request body").Error())
+	}
+	if req.UID == "" || req.PID == "" {
+		return c.String(http.StatusBadRequest, "uid and idx fields are both required")
+	}
+
+	u, err := c.LoadUser(c.Request().Context(), req.UID)
+	if err != nil{
+		return err
+	}
+
+	// get pid to delete then remove the entry
+	idx := 0
+	for i, p := range u.Programs {
+		if p == req.PID {
+			idx = i
+			break
+		}
+	}
+	if idx >= len(u.Programs) {
+		return errors.New("invalid PID")
+	}
+	toDelete := u.Programs[idx]
+	u.Programs = append(u.Programs[:idx], u.Programs[idx+1:]...)
+
+	// pref := d.Collection(programsPath).Doc(toDelete)
+	c.StoreUser(c.Request().Context(), u)
```

Comment:
```suggestion
	err = c.StoreUser(c.Request().Context(), u)
	if err != nil {
		return err
        }
```
Something quick like this should make the linter happy